src/main/java/it/gov/pagopa/apiconfig/selfcareintegration/repository/ExtendedChannelRepository.java

```diff
@@ -0,0 +1,32 @@
+package it.gov.pagopa.apiconfig.selfcareintegration.repository;
+
+import it.gov.pagopa.apiconfig.starter.entity.Canali;
+import it.gov.pagopa.apiconfig.starter.repository.CanaliRepository;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+@SuppressWarnings(
+    "java:S100") // Disabled naming convention rule for method name to use Spring Data interface
+@Repository
+public interface ExtendedChannelRepository extends CanaliRepository {
+
+  @Query(
+      value =
+          "SELECT DISTINCT c FROM Canali c WHERE (c.fkIntermediarioPsp.objId = :fkIntermediario) ORDER"
+              + " BY c.idCanale")
```

Comment:
```suggestion
```
use spring data feature (with the name of the methods) or [specifications](https://www.baeldung.com/rest-api-search-language-spring-data-specifications)

Reply:
nice tip, updated this repository and the `ExtendedCreditorInstitutionStationRepository` one